etc/ci-download.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+#
+# This script expects a single argument, which should be an URL pointing to a
+# file for download; it then tries to download that file, into a local file
+# with the same name as the remote file.
+# If a file of the same name exists in ~/.gap-package-cache, then this file
+# is used if it has the same hash
+set -e
+
+PACKAGEDIR=~/.gap-package-cache
+
+name=${*##*/}
+# Check if file already exists and has right hash
+if [ -f ${PACKAGEDIR}/"${name}" ] ; then
+    command -v sha256sum > /dev/null 2>&1 &&
+    {
+        hash1=$(curl --retry 5 --retry-delay 5 --max-time 10 -L "$*.sha256")
+        hash2=$(sha256sum ${PACKAGEDIR}/"${name}" | cut -d' ' -f1)
+        if [ "x$hash1" == "x$hash2" ] ; then
+            echo "Getting file from cache"
+            cp ${PACKAGEDIR}/"${name}" .
+            exit 0
+        fi
+    }
+fi
+
+if curl -L --retry 5 --retry-delay 5 --max-time 120 -O "$@" ; then
```

Comment:
Why $@ here but $* above

Reply:
Honestly, shellshock told me using "$@.sha256" was bad as it was combining an array and a string, so I did what I was told (I had a look at it seemed reasonable). I can change this one too. Certainly it should probably be $* now, as I don't claim this script would work downloading multiple files any more (if it ever did)